systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java

```diff
@@ -241,4 +246,47 @@ public static void waitForSpecificLabelKeyValue(String secretName, String namesp
                 )
         );
     }
+
+    /**
+     * To test tls-external authentication, client needs to have an external certificates provided
+     * To simulate externally provided certs, these steps are done:
+     * 1. Generate private key and csr (containing at least common name CN) for user
+     * 2. Generate a certificate by signing CSR using private key from secret generated for clients by operator
+     * 3. Create user secret from private key, generated certificate and certificate from secret created for clients by operator
+     * @param namespaceName Namespace name where kafka is deployed
+     * @param userName User name of kafkaUser
+     * @param clusterName Name of cluster which kafkaUser belongs to
+     */
+    public static void createExternalTlsUserSecret(String namespaceName, String userName, String clusterName) {
+        try {
+            File clientPrivateKey = OpenSsl.generatePrivateKey();
+
+            File csr = OpenSsl.generateCertSigningRequest(clientPrivateKey, "/CN=" + userName);
+            String caCrt = KubeClusterResource.kubeClient(namespaceName).getSecret(
+                KafkaResources.clientsCaCertificateSecretName(clusterName)).getData().get("ca.crt");
+            String caKey = KubeClusterResource.kubeClient(namespaceName).getSecret(KafkaResources.clientsCaKeySecretName(clusterName)).getData().get("ca.key");
+
+            File clientCert = OpenSsl.generateSignedCert(csr,
+                                                         SystemTestCertManager.exportCaDataToFile(new String(Base64.getDecoder().decode(caCrt), StandardCharsets.UTF_8), "ca", ".crt"),
+                                                         SystemTestCertManager.exportCaDataToFile(new String(Base64.getDecoder().decode(caKey), StandardCharsets.UTF_8), "ca", ".key"));
+
+            Secret secretBuilder = new SecretBuilder()
+                .withApiVersion("v1")
+                .withKind("Secret")
```

Comment:
Can we somehow use constants and void using magic constants like this?

Reply:
We have constants for that IIRC

Reply:
it is not needed to be specified, as the secret gets created with the builder even without the specified api and kind